In [1]:
using Symbolics
using Latexify

In [2]:
@variables g00, g11, g01, ν, r0, r1, t00, t01, t11, u0, u1, μ, T, γ, v, χ₁, χ₂, χ₀ ;

In [3]:
A00021 = 12χ₂*u0/(μ^4 *T)*(19//3*ν*(g00 + 4u0*u0)/μ + 14/T/μ *r0*u0 - t00)

(12//1)*u0*χ₂*((19//3)*ν*(g00 + 4(u0^2))*(μ^-1) + 14r0*u0*(T^-1)*(μ^-1) - t00)*(T^-1)*(μ^-4)

In [4]:
A00022 = -24χ₂/μ^4 *(ν/μ/T*(2//3*g00*u0 -1//2*g00*u0 - 4//3*u0^3) + r0*(g00 - u0^2 - (u0^2 + g00//2)) + t00 * u0/T)

-24χ₂*(r0*((1//2)*g00 - (2(u0^2))) + t00*u0*(T^-1) + ν*(T^-1)*(μ^-1)*((1//6)*g00*u0 - ((4//3)*(u0^3))))*(μ^-4)

In [5]:
A00023 = -96χ₂/μ^5 * (ν/T*(- u0*g00/2 + u0^3 ) + r0*μ*(g00/4 + u0^2))

-96χ₂*(μ^-5)*(r0*μ*(0.25g00 + u0^2) + ν*(T^-1)*(u0^3 - (0.5g00*u0)))

In [6]:
A0002 = simplify(A00021 + A00022 + A00023; expand=true,
            threaded=false,
            thread_subtree_cutoff=100,
            rewriter=nothing)

(168//1)*r0*χ₂*(T^-2)*(u0^2)*(μ^-5) + (240//1)*ν*χ₂*(T^-1)*(u0^3)*(μ^-5) + 120.0g00*u0*ν*χ₂*(T^-1)*(μ^-5) - (36.0g00*r0*χ₂*(μ^-4)) - (48r0*χ₂*(u0^2)*(μ^-4)) - ((36//1)*t00*u0*χ₂*(T^-1)*(μ^-4))

In [7]:
A0002_s=simplify(substitute(A0002, Dict([g00 => -1, g11 => 1, μ => -T^(-2), r0 => r1*v, u0 => γ, t00 => v^2*t11])))

36.0r1*v*χ₂*(T^8) + 120.0γ*ν*χ₂*(T^9) - (240.0ν*χ₂*(T^9)*(γ^3)) - (216.0r1*v*χ₂*(T^8)*(γ^2)) - ((36//1)*t11*γ*χ₂*(T^7)*(v^2))

$A^{000}_2 = \frac{\chi_2}{\mu^4}(\nu \gamma T(120 - 240\gamma^2) + r^1 v(36 - 216\gamma^2) - 36v^2\gamma \tilde{\tau}^{11}/T)$

In [8]:
A01021 = 12χ₂*u0/(μ^6 *T)*(19//3*ν*μ*(g01 + 4u1*u0) + 7μ/T *(r0*u1+r1*u0) - μ^2*t01)

(12//1)*u0*χ₂*((19//3)*μ*ν*(g01 + 4u0*u1) + 7μ*(T^-1)*(r0*u1 + r1*u0) - (t01*(μ^2)))*(T^-1)*(μ^-6)

In [9]:
A01022 = -24χ₂/μ^4 *(ν/μ/T*(1//3*g01*u0 + 1//3*g00*u1 -1//2*g01*u0 - 4//3*u0^2*u1) 
    + r0*g01//2 + r1*g00//2 - u0^2*r1//2 - u0*u1*r0//2 - r0*(u0*u1 + g01//2)
+ t01 * u0/T//2 + t00 * u1/T//2)

-24χ₂*((1//2)*g00*r1 + (1//2)*g01*r0 + (2//1)*t00*u1*(T^-1) + (2//1)*t01*u0*(T^-1) + ν*((1//3)*g00*u1 - ((1//6)*g01*u0) - ((4//3)*u1*(u0^2)))*(T^-1)*(μ^-1) - (r0*((1//2)*g01 + u0*u1)) - ((1//2)*r1*(u0^2)) - ((1//2)*r0*u0*u1))*(μ^-4)

In [10]:
A01023 = -96χ₂/μ^5 * (ν/T*(u0*g01/2 + u0^2*u1 - 1//2*g00*u1 - 1//2*g01*u0) 
    + r0*μ*(g01/4 + u0*u1))

-96χ₂*(μ^-5)*(r0*μ*(0.25g01 + u0*u1) + ν*(T^-1)*(u1*(u0^2) - ((1//2)*g00*u1)))

In [11]:
A0102 = simplify(A01021 + A01022 + A01023; expand=true,
            threaded=false,
            thread_subtree_cutoff=100,
            rewriter=nothing)

(12//1)*r1*χ₂*(u0^2)*(μ^-4) + (84//1)*r1*χ₂*(T^-2)*(u0^2)*(μ^-5) + (40//1)*g00*u1*ν*χ₂*(T^-1)*(μ^-5) + (80//1)*g01*u0*ν*χ₂*(T^-1)*(μ^-5) + (84//1)*r0*u0*u1*χ₂*(T^-2)*(μ^-5) + (240//1)*u1*ν*χ₂*(T^-1)*(u0^2)*(μ^-5) - ((12//1)*g00*r1*χ₂*(μ^-4)) - (24.0g01*r0*χ₂*(μ^-4)) - ((60//1)*r0*u0*u1*χ₂*(μ^-4)) - ((48//1)*t00*u1*χ₂*(T^-1)*(μ^-4)) - ((60//1)*t01*u0*χ₂*(T^-1)*(μ^-4))

In [12]:
A0102_s=simplify(substitute(A0102, Dict([g00 => -1, g11 => 1, g01 => 0, μ => -T^(-2), r0 => r1*v, u0 => γ, u1 => γ*v, t00 => v^2*t11, t01 => v*t11 ])))

(12//1)*r1*χ₂*(T^8) + 40.0v*γ*ν*χ₂*(T^9) - (72.0r1*χ₂*(T^8)*(γ^2)) - (144.0r1*χ₂*(T^8)*(v^2)*(γ^2)) - ((60//1)*t11*v*γ*χ₂*(T^7)) - ((48//1)*t11*γ*χ₂*(T^7)*(v^3)) - (240.0v*ν*χ₂*(T^9)*(γ^3))

$A^{001}_2 = \chi_2*\mu^4(\nu T v \gamma(40 - 240\gamma^2) + r^1(12  - 72\gamma^2 - 144\gamma^2 v^2) - \tau^{11}v\gamma/T(60 + 48v^2)$

In [13]:
A01121 = 12χ₂*u0/(μ^6 *T)*(19//3*ν*μ*(g11 + 4u1*u1) + 7μ/T *(r1*u1+r1*u1) 
    - μ^2*t11)

(12//1)*u0*χ₂*((19//3)*μ*ν*(g11 + 4(u1^2)) + 14r1*u1*μ*(T^-1) - (t11*(μ^2)))*(T^-1)*(μ^-6)

In [14]:
A01122 = -24χ₂/μ^4 *(ν/μ/T*(1//3*g01*u1 + 1//3*g01*u1 -1//2*g11*u0 - 4//3*u1^2*u0) 
    + r1*g01//2 + r1*g01//2 - u0*u1*r1//2 - u0*u1*r1//2 - r0*(u1*u1 + g11//2)
+ t01 * u1/T//2 + t01 * u1/T//2)

-24χ₂*(g01*r1 + (4//1)*t01*u1*(T^-1) + ν*((2//3)*g01*u1 - ((1//2)*g11*u0) - ((4//3)*u0*(u1^2)))*(T^-1)*(μ^-1) - (r0*((1//2)*g11 + u1^2)) - (r1*u0*u1))*(μ^-4)

In [15]:
A01123 = -96χ₂/μ^5 * (ν/T*(u0*g11 + u0*u1^2 - 1//2*g01*u1 - 1//2*g01*u1) 
    + r0*μ*(g11/4 + u1*u1))

-96χ₂*(μ^-5)*(r0*μ*(0.25g11 + u1^2) + ν*(g11*u0 + u0*(u1^2) - (g01*u1))*(T^-1))

In [16]:
A0112 = simplify(A01121 + A01122 + A01123; expand=true,
            threaded=false,
            thread_subtree_cutoff=100,
            rewriter=nothing)

(24//1)*r1*u0*u1*χ₂*(μ^-4) + (80//1)*g01*u1*ν*χ₂*(T^-1)*(μ^-5) + (168//1)*r1*u0*u1*χ₂*(T^-2)*(μ^-5) + (240//1)*u0*ν*χ₂*(T^-1)*(u1^2)*(μ^-5) - (24g01*r1*χ₂*(μ^-4)) - (12.0g11*r0*χ₂*(μ^-4)) - (72r0*χ₂*(u1^2)*(μ^-4)) - ((96//1)*t01*u1*χ₂*(T^-1)*(μ^-4)) - ((12//1)*t11*u0*χ₂*(T^-1)*(μ^-4)) - ((8//1)*g11*u0*ν*χ₂*(T^-1)*(μ^-5))

In [17]:
A0112_s=simplify(substitute(A0112, Dict([g00 => -1, g11 => 1, g01 => 0, μ => -T^(-2), r0 => r1*v, u0 => γ, u1 => γ*v, t00 => v^2*t11, t01 => v*t11 ])))

8.0γ*ν*χ₂*(T^9) - (12.0r1*v*χ₂*(T^8)) - ((12//1)*t11*γ*χ₂*(T^7)) - (144.0r1*v*χ₂*(T^8)*(γ^2)) - (72r1*χ₂*(T^8)*(v^3)*(γ^2)) - ((96//1)*t11*γ*χ₂*(T^7)*(v^2)) - (240.0ν*χ₂*(T^9)*(v^2)*(γ^3))

$A^{011}_2 = \frac{\chi_2}{\mu^4}(\nu T \gamma(8 - 240\gamma^2) + r^1v(-12 -144\gamma^2 - 72\gamma^2 v^2) + \frac{\tilde{\tau}^{11}\gamma}{T}(-12 - 96v^2))$



In [18]:
$A^{011}_2 = \frac{\chi_2}{\mu^4}(\nu T \gamma(8 - 240\gamma^2) + r^1v(-12 -144\gamma^2 - 72\gamma^2 v^2) + \frac{\tilde{\tau}^{11}\gamma}{T}(-12 - 96v^2))$



LoadError: syntax: "\" is not a unary operator

## De aqui es la cuenta correcta (lo anterior no se si está bien)

In [19]:
DIC = Dict([g00 => -1, g11 => 1, g01 => 0, μ => -T^(-2), r0 => r1*v, u0 => γ, u1 => γ*v, t00 => v^2*t11, t01 => v*t11])

Dict{Num, Num} with 9 entries:
  μ   => -(T^-2)
  g01 => 0
  u1  => v*γ
  g00 => -1
  g11 => 1
  r0  => r1*v
  t01 => t11*v
  u0  => γ
  t00 => t11*(v^2)

### El segundo orden es:

\begin{eqnarray*}
    A_2^{abc} &=& -\frac{12\chi_2}{\mu^4}[\frac{10}{3}\frac{\nu}{\sqrt{-\mu}}u^a g^{bc}
    + \frac{20\nu}{\sqrt{-\mu}} u^a u^b u^c  + \frac{20}{3}\frac{\nu}{\sqrt{-\mu}}g^{a(b}u^{c)} \\
    &+& 12 u^a u^{(b}r^{c)} + 2g^{a(b}r^{c)} + 6r^a u^b u^c 
    + r^a g^{bc} \\
    &+& 2\sqrt{-\mu}\tilde{\tau}^{a(b}u^{c)} + \sqrt{-\mu}u^a\tilde{\tau}^{bc}]
\end{eqnarray*}

In [20]:
A111_2 = -12χ₂*(ν*T*(10*u1*g11 + 10*g11*u1 + 10*g11*u1 + 60u1*u1*u1)//3 
    + (6r1*u1*u1 + 6r1*u1*u1 + g11*r1 + g11*r1 + 6r1*u1*u1 
        + r1*g11) + (t11*u1 + t11*u1 + u1*t11)/T)

-12χ₂*(T*ν*((20//1)*(u1^3) + (10//1)*g11*u1) + 3g11*r1 + 18r1*(u1^2) + 3t11*u1*(T^-1))

In [21]:
A011_2 = -12χ₂*(ν*T*(10*u0*g11 + 10*g01*u1 + 10*g01*u1 + 60u0*u1*u1)//3 
    + (6r1*u1*u0 + 6r1*u0*u1 + g01*r1 + g01*r1 + 6r0*u1*u1 
        + r0*g11) + (t01*u1 + t01*u1 + u0*t11)/T)

-12χ₂*(g11*r0 + (T^-1)*(t11*u0 + 2t01*u1) + T*ν*((20//3)*g01*u1 + (10//3)*g11*u0 + (20//1)*u0*(u1^2)) + 2g01*r1 + 6r0*(u1^2) + 12r1*u0*u1)

In [22]:
A001_2 = -12χ₂*(ν*T*(10*u0*g01 + 10*g00*u1 + 10*g01*u0  + 60u0*u0*u1)//3 
    + (6r1*u0*u0 + 6r0*u0*u1 + g00*r1 + g01*r0 + 6r0*u0*u1 
        + r0*g01) + (t00*u1 + t01*u0 + u0*t01)/T)

-12χ₂*(g00*r1 + (T^-1)*(t00*u1 + 2t01*u0) + T*ν*((10//3)*g00*u1 + (20//3)*g01*u0 + (20//1)*u1*(u0^2)) + 2g01*r0 + 6r1*(u0^2) + 12r0*u0*u1)

In [23]:
A000_2 = -12χ₂*(ν*T*(10*u0*g00 + 20*g00*u0 + 60u0*u0*u0)//3 
    + (6r0*u0*u0 + 6r0*u0*u0 + g00*r0 + g00*r0 + 6r0*u0*u0 
        + r0*g00) + (t00*u0 + t00*u0 + u0*t00)/T)

-12χ₂*(T*ν*((20//1)*(u0^3) + (10//1)*g00*u0) + 3g00*r0 + 18r0*(u0^2) + 3t00*u0*(T^-1))

In [24]:
simplify(substitute(A000_2, Dict(DIC)))

-12χ₂*(T*ν*((20//1)*(γ^3) - ((10//1)*γ)) + 18r1*v*(γ^2) + 3t11*γ*(T^-1)*(v^2) - (3r1*v))

In [25]:
simplify(substitute(A001_2, Dict(DIC)))

-12χ₂*((T^-1)*(t11*γ*(v^3) + 2t11*v*γ) + T*ν*((20//1)*v*(γ^3) - ((10//3)*v*γ)) + 6r1*(γ^2) + 12r1*(v^2)*(γ^2) - r1)

In [26]:
simplify(substitute(A011_2, Dict(DIC)))

-12χ₂*(r1*v + (T^-1)*(t11*γ + 2t11*γ*(v^2)) + T*ν*((10//3)*γ + (20//1)*(v^2)*(γ^3)) + 12r1*v*(γ^2) + 6r1*(v^3)*(γ^2))

In [27]:
simplify(substitute(A111_2, Dict(DIC)))

-12χ₂*(3r1 + T*ν*((10//1)*v*γ + (20//1)*(v^3)*(γ^3)) + 18r1*(v^2)*(γ^2) + 3t11*v*γ*(T^-1))

### El primer orden es:

$$
    A_1{}^{abc}=\frac{\chi_1\sqrt{-\mu}}{\mu^3}\left[2h^{a(b}u^{c)} + u^ah^{bc} + 3u^a u^b u^c\right] = \frac{\chi_1\sqrt{-\mu}}{\mu^3}\left[g^{ab}u^c + g^{ac}u^b + g^{bc}u^a + 6u^a u^b u^c\right]
$$

In [28]:
A111_1 = χ₁/T/μ^3*[g11*u1 + g11*u1 + g11*u1 + 6u1*u1*u1]

1-element Vector{Num}:
 χ₁*(T^-1)*(μ^-3)*(6(u1^3) + 3g11*u1)

In [29]:
simplify(substitute(A111_1, Dict([g00 => -1, g11 => 1, g01 => 0, μ => -T^(-2), r0 => r1*v, u0 => γ, u1 => γ*v, t00 => v^2*t11, t01 => v*t11])))

1-element Vector{Num}:
 χ₁*(T^-1)*(μ^-3)*(6(u1^3) + 3g11*u1)

In [30]:
A011_1 = χ₁/T/μ^3*[g01*u1 + g01*u1 + g11*u0 + 6u1*u1*u0]

1-element Vector{Num}:
 χ₁*(g11*u0 + 2g01*u1 + 6u0*(u1^2))*(T^-1)*(μ^-3)

In [31]:
simplify(substitute(A011_1, Dict(DIC)))

1-element Vector{Num}:
 χ₁*(g11*u0 + 2g01*u1 + 6u0*(u1^2))*(T^-1)*(μ^-3)

In [32]:
A001_1 = χ₁/T/μ^3*[g00*u1 + g01*u0 + g01*u0 + 6u1*u0*u0]

1-element Vector{Num}:
 χ₁*(g00*u1 + 2g01*u0 + 6u1*(u0^2))*(T^-1)*(μ^-3)

In [33]:
simplify(substitute(A001_1, Dict(DIC)))

1-element Vector{Num}:
 χ₁*(g00*u1 + 2g01*u0 + 6u1*(u0^2))*(T^-1)*(μ^-3)

In [34]:
A000_1 = χ₁/T/μ^3*[g00*u0 + g00*u0 + g00*u0 + 6u0*u0*u0]

1-element Vector{Num}:
 χ₁*(T^-1)*(μ^-3)*(6(u0^3) + 3g00*u0)

In [35]:
simplify(substitute(A000_1, Dict(DIC)))

1-element Vector{Num}:
 χ₁*(T^-1)*(μ^-3)*(6(u0^3) + 3g00*u0)